sdktests/testapi_sdk_client.go

```diff
@@ -107,7 +110,16 @@ type SDKClient struct {
 // The object's lifecycle is tied to the test scope that created it; it will be automatically closed
 // when this test scope exits. It can be reused by subtests until then.
 func NewSDKClient(t *ldtest.T, configurers ...SDKConfigurer) *SDKClient {
+	count := atomic.AddInt32(&currentlyExistingClientInstances, 1)
+	if count == 2 && t.Capabilities().Has(servicedef.CapabilitySingleton) {
+		atomic.AddInt32(&currentlyExistingClientInstances, -1)
+		t.Errorf("Test tried to create an SDK client instance when one already existed, and this SDK is a singleton")
```

Comment:
The test service would have failed anyway when we tried to create another client, if it doesn't support multiple clients. But failing here is faster.